R/civis_ml_utils.R

```diff
@@ -19,19 +19,93 @@ CIVIS_ML_CLASSIFIERS <- c("sparse_logistic",
                           "multilayer_perceptron_classifier",
                           "stacking_classifier")
 
-CIVIS_ML_TEMPLATE_IDS <- data.frame(
-  id = c(9112, 9113, 9968, 9969, 10582, 10583, 11219, 11220, 11221),
-  version = c(1.1, 1.1, 2.0, 2.0, 2.1, 2.1, 2.2, 2.2, 2.2),
-  name = c("train", "predict", "train", "predict", "train", "predict", "train", "predict", "register"),
-  stringsAsFactors = FALSE
-)
+
+get_template_ids_all_versions <- function(){
+  # Get template IDs for all accessible CivisML versions.
+
+  template_alias_objects <- fetch_until(aliases_list,
+                                       limit=1000,
+                                       object_type='template_script',
+                                       function(x) length(x) == 0)
+
+
+  template_aliases <- sapply(template_alias_objects, `[[`, 'alias')
+
+  # Template script aliases are not used exclusively for CivisML
+  civisml_template_alias_objects <- template_alias_objects[startsWith(template_aliases,
+                                                                      'civis-civisml-')]
+
+  ids <- lapply(civisml_template_alias_objects, function(alias_obj){
+
+        id = alias_obj$objectId
+
+        job_type_version = get_job_type_version(alias_obj$alias)
+
+        return(list(id = id,
+                    version = job_type_version$version,
+                    name = job_type_version$job_type))
+
+    })
+
+  CIVIS_ML_TEMPLATE_IDS <- data.frame(matrix(unlist(ids),
```

Comment:
Not entirely sure, but is R cool with naming intermediate vars in ALL_CAPS? It looks like `CIVIS_ML_TEMPLATE_IDS` was named as such before this PR because it's a constant, but I might be missing something.

Reply:
Agreed, since this isn't a package level constant anymore, it doesn't need to be in all caps. Not a syntax requirement, but a nice convention for readability.

Reply:
Doesn't need to be in all caps

Reply:
Changed to lower case. Thanks! Resolving this convo